packages/bento-design-system/src/DateField/Input.tsx

```diff
@@ -0,0 +1,142 @@
+import { useTextField } from "@react-aria/textfield";
+import { InputHTMLAttributes, RefObject, useEffect, useState } from "react";
+import { Box } from "../internal";
+import { input } from "./DateField.css";
+import { getInputValue, parseDate as _parseDate } from "@datepicker-react/hooks";
+import { useDateFormatter } from "@react-aria/i18n";
+import InputMask from "react-input-mask";
+
+type Props = {
+  for: "startDate" | "endDate";
+  currentDate: Date | null;
+  inputRef: RefObject<HTMLInputElement>;
+  isFocused: boolean;
+  onClick: () => void;
+  onFocus: () => void;
+  onBlur: () => void;
+  onChange: (date: Date) => void;
+  disabled: boolean;
+  readOnly: boolean;
+  onDateSelect: (date: Date) => void;
+  onDateClear: () => void;
+  isOpen: boolean;
+};
+
+function useDatePattern() {
+  const dateFormatter = useDateFormatter();
+  const parts: any[] = dateFormatter.formatToParts();
+
+  return parts
+    .map<string | undefined>((part) => {
+      switch (part.type) {
+        case "day":
+          return "dd";
+        case "month":
+          return "MM";
+        case "year":
+          return "yyyy";
+        default:
+          return undefined;
+      }
+    })
+    .filter((a) => !!a)
+    .join("/");
+}
+
+function parseDate(value: string, pattern: string): Date | null {
+  const parsedDate = _parseDate(value, pattern, new Date());
+  if (!isNaN(parsedDate.getTime())) {
+    return parsedDate;
+  }
+  return null;
+}
+
+export function Input(props: Props) {
+  const dateFormatter = useDateFormatter({ day: "2-digit", month: "2-digit", year: "numeric" });
```

Comment:
sono un po' confuso: qui chiamiamo `useDateFormatter` con degli argomenti, mentre sopra in `useDatePattern` lo chiamiamo senza argomenti. Come mai non è lo stesso `dateFormatter`?